lms/services/oauth2_token.py

```diff
@@ -69,7 +69,9 @@ def get(self, service=Service.LMS):
             ) from err
 
 
-def oauth2_token_service_factory(_context, request):
+def oauth2_token_service_factory(_context, request, user_id: str | None = None):
     return OAuth2TokenService(
-        request.db, request.lti_user.application_instance, request.lti_user.user_id
+        request.db,
+        request.lti_user.application_instance,
```

Comment:
> That being said, depending on the service that could be costly refactor.

At first glance it looks mostly like a tedious refactor rather than complex one.

Reply:
My point is that these should not be service arguments but extra service method parameters, ie making this not the "Current user token service" but "User token service".

I reckon in this case, as this is the first time we have this need the refactor will look much more verbose as this ties into the HTTPService as well. Calling the service factory does the trick thought.